File: MPChartLib/src/main/java/com/github/mikephil/charting/interfaces/LineDataProvider.java
package com.github.mikephil.charting.interfaces;

import com.github.mikephil.charting.components.YAxis;
import com.github.mikephil.charting.data.LineData;

public interface LineDataProvider extends BarLineScatterCandleBubbleDataProvider {

    LineData getLineData();

    YAxis getAxis(YAxis.AxisDependency dependency);
}

File: MPChartLib/src/main/java/com/github/mikephil/charting/interfaces/dataprovider/LineDataProvider.kt
package com.github.mikephil.charting.interfaces.dataprovider

import com.github.mikephil.charting.components.YAxis
import com.github.mikephil.charting.components.YAxis.AxisDependency
import com.github.mikephil.charting.data.LineData

interface LineDataProvider : BarLineScatterCandleBubbleDataProvider {
    val lineData: LineData
    fun getAxis(dependency: AxisDependency): YAxis?
}